bugrap/src/main/java/org/vaadin/bugrap/ui/ReportsModel.java

```diff
@@ -93,11 +94,42 @@ public void saveReport() throws ValidationException {
 			getRepository().save(report);
 		}
 	}
+	
+	public void save() {
+		Report report = getReportById(reportForEdit.getId());
+		ReportUtil.setFields(report, reportForEdit);
+		getRepository().save(report);
+	}
 
 	public Report getReportById(long reportId) {
 		return getRepository().getReportById(reportId);
 	}
 
+	public void openReportDetail(long reportId) {
+		getNavigator().navigateTo(NAV_REPORTDET + "/reportId="+ reportId);
```

Comment:
Formatting issue.

Why are you navigating to the report detail? Isn't it just a matter of showing it?